github/github.go

```diff
@@ -390,7 +390,7 @@ func parseRate(r *http.Response) Rate {
 // The provided ctx must be non-nil. If it is canceled or times out,
 // ctx.Err() will be returned.
 func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Response, error) {
-	req = req.WithContext(ctx)
+	ctx, req = addContext(ctx, req)
```

Comment:
I see. To make my question more specific, how does it have an effect given that the returned modified value of `ctx` is never used (aside from `ctx.Done()` and `ctx.Err()`) and never returned from `Do`?

Is it because the implementation of `appengine.WithContext` has side effects, other than just returning a modified copy of `ctx`?